[Term Entry] Adding Pytorch Tensor zeros (#4971)
Codecademy/docs: content/pytorch/concepts/tensors/terms/zeros/zeros.md

> ---
> Title: 'ZEROS'
> Description: 'Initializing a new tensor with zeros.'
> Subjects:
>   - 'Data Science'
>   - 'Machine Learning'
>   - 'AI'
> Tags:
>   - 'AI'
>   - 'Data Structures'
>   - 'Arrays'
>   - 'Deep Learning'
> CatalogContent:
>   - 'intro-to-py-torch-and-neural-networks'
>   - 'py-torch-for-classification'
>   - 'paths/machine-learning'
> ---

Reviewer: ```suggestion
---
Title: '.zeros()'
Description: 'Initializes a new tensor filled with zeros, with a specified shape.'
Subjects:
  - 'AI'
  - 'Data Science'
  - 'Machine Learning'
Tags:
  - 'AI'
  - 'Arrays'
  - 'Data Structures'
  - 'Deep Learning'
CatalogContent:
  - 'intro-to-py-torch-and-neural-networks'
  - 'py-torch-for-classification'
  - 'paths/data-science'
---
```

Reviewer: Modifications:
1. title is `.zeros`
2. description should start with verbs
3. Tags and subjects should be in alphabetical order

> The `.zeros()` method returns a [Tensor](https://www.codecademy.com/resources/docs/pytorch/tensors) of a specified shape with zeros. Optionally, the output tensor, datatype, layout, device, and whether autograd records operations can be specified.

Reviewer: ```suggestion
The **`.zeros()`** method returns a tensor of a specified shape filled with zeros. Optionally, the output tensor, its datatype, layout, device, and whether autograd records operations can be selected.
```

Reviewer: Modifications: 
1. the first instance of a term should be in bold and backticks
2. There is no entry of tensors as of now, so removed the link

## Syntax

> ```py

Reviewer: ```suggestion
```pseudo
```

Reviewer: Syntax is written using the pseudo block

> import torch
> 
> torch.zeros(shape)
> 
> # with defaults explicitly stated

Reviewer: ```suggestion
```

Reviewer: Can be removed

> torch.zeros(shape, out=None, dtype=torch.float32, layout=torch.strided, device=torch.device('cpu'), requires_grad=False)

Reviewer: ```suggestion
torch.zeros(size, out=None, dtype=None, layout=torch.strided, device=None, requires_grad=False)
```

Reviewer: Reefrence: https://pytorch.org/docs/stable/generated/torch.zeros.html

Author: I see. I saw that in the description it said that if the dtype is written as None, the global default will be assigned, which, if not assigned, is torch.float32. Should we still leave it as none?

```

The shape is specified as a variable, tuple, or list of integers. Optional parameters are as follows:

- `out`: The output Tensor, defaults to None.
- `dtype`: Datatype (`torch.dtype`) of the zeros, defaults to the global default.
- `layout`: Layout (`torch.layout`) of the output Tensor, defaults to `torch.strided`.
- `device`: Device (`torch.device`) of the output Tensor, defaults to the global default.

> - `requires_grad`: Boolean of whether autograd will record operations on the output Tensor, defaults to False.

Reviewer: Please refer to the link: https://pytorch.org/docs/stable/generated/torch.zeros.html

only the `layout` parameter has a default value other than global, all others have `None` as their default. Please reframe the parameter descriptions. Also add a bullet point for the `size` parameter

## Example

```python
import torch

t0 = torch.zeros(3)

> t1 = torch.zeros((4, 2), dtype=torch.int16)

Reviewer: Could you add comments here on what is happening in these two lines?

print(t0)
print(t1)
```

The returned tensors are as follows:

```shell

> tensor([ 0.,  0., 0.])
> 
> tensor([[ 0,  0],
>         [ 0,  0],
>         [ 0,  0],
>         [ 0,  0]])

Reviewer: ```suggestion
tensor([0., 0., 0.])
tensor([[0, 0],
        [0, 0],
        [0, 0],
        [0, 0]], dtype=torch.int16)
```

---
Title: '.zeros()'
Description: 'Initializes a new tensor filled with zeros, with a specified shape.'
Subjects:
  - 'AI'
  - 'Data Science'
  - 'Machine Learning'
Tags:
  - 'AI'
  - 'Arrays'
  - 'Data Structures'
  - 'Deep Learning'
CatalogContent:
  - 'intro-to-py-torch-and-neural-networks'
  - 'py-torch-for-classification'

>   - 'paths/data-science'

Reviewer 2: ``` suggestion
CatalogContent:
  - 'intro-to-py-torch-and-neural-networks'
  - 'py-torch-for-classification'
  - 'paths/data-science'
```
Only use the course/path that is already present in the [catalog-content. md file ](documentation/catalog-content.md)and else if you do not have the paths that you want to include in the PR present in the list then add one.

Author: sorry i don't see the file, is there another link i can see what courses there are?

Author: oh i found it sorry nvm

---

> The **`.zeros()`** method returns a tensor of a specified shape filled with zeros. Optionally, the output tensor, its datatype, layout, device, and whether autograd records operations can be selected.

Reviewer 2: ```suggestion
The **`.zeros()`** method returns a tensor of a specified shape filled with zeros. Optionally, the output tensor, its datatype, layout, device, and whether autograd records operations can be selected.
```
You can rewrite this sentence for more readability.

## Syntax

```pseudo
torch.zeros(size, out=None, dtype=None, layout=torch.strided, device=None, requires_grad=False)
```

The parameters are as follows:

- `size`: The shape of the tensor, specified as a variable, tuple, or list of integers.
- `out`: The output Tensor, defaults to `None`.
- `dtype`: The datatype (`torch.dtype`) of the zeros, defaults to `None`.
- `layout`: The layout (`torch.layout`) of the output tensor, defaults to `torch.strided`.
- `device`: The device (`torch.device`) of the output tensor, defaults to `None`.
- `requires_grad`: A boolean indicating whether autograd will record operations on the output tensor, defaults to `False`.

## Example

```py

> import torch
> 
> # Define a tensor with three values in one row
> t0 = torch.zeros(3)
> 
> # Define a tensor with four rows, two values per row. All values have the datatype torch.int16
> t1 = torch.zeros((4, 2), dtype=torch.int16)
> 
> print(t0)
> print(t1)

Reviewer 2: ```suggestion
import torch

# Define a tensor with three values in one row
t0 = torch.zeros(3)

# Define a tensor with four rows, two values per row. All values have the datatype torch.int16
t1 = torch.zeros((4, 2), dtype=torch.int16)

print(t0)
print(t1)
```

Author: I am unclear as to what has changed here?
